openupgradelib/openupgrade_140.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+"""This module provides simple tools for OpenUpgrade migration, specific for
+the 13.0 -> 14.0 migration.
+"""
+
+import logging
+from itertools import product
+
+from psycopg2.extensions import AsIs
+
+from .openupgrade import update_field_multilang
+from .openupgrade_tools import (
+    convert_html_fragment,
+    convert_html_replacement_class_shortcut as _r,
+)
+
+logger = logging.getLogger("OpenUpgrade")
+logger.setLevel(logging.DEBUG)
+
+_PREFIX = ["bg-", "text-", "border-"]
+_CONTEXTS = (
+    ("alpha", 1),
+    ("beta", 2),
+    ("gamma", 3),
+    ("delta", 4),
+    ("epsilon", 5),
+    ("zeta", 6),
+    ("eta", 7),
+)
+
+# These replacements are specific for Odoo v13 to v14
+_ODOO14_REPLACEMENTS = (
+    *(
+        _r(f"{prefix}{v13}", f"{prefix}o-color-{v14}")
+        for prefix, (v13, v14) in product(_PREFIX, _CONTEXTS)
+    ),
+    *(_r(f"btn-{v13}", f"bg-o-color-{v14}") for (v13, v14) in _CONTEXTS),
+    *(_r(f"o-mm-def-color-{v13}", f"o-mm-def-color-{v14}") for (v13, v14) in _CONTEXTS),
+    *(_r(f"btn-outline-{v13}", f"border-o-color-{v14}") for (v13, v14) in _CONTEXTS),
```

Comment:
I see `btn-outline-alpha` converted to `btn-primary` here:

https://github.com/odoo/odoo/commit/7736cde22c3bce06305045a818a2d70892cf25ef#diff-a86d102c3706a1fe6bccf059dd6ebc3b44069375110ea7d338cef52475fae51aR386

Is the usual rule?

Reply:
btn-outline-alpha applies a coloured border and to follow the same style the appropriate class is "border-o-color-", it seems to be an isolated case of changing the button style completely using btn-primary.
The intention is to remove deprecated classes without changing their styling, one-off design changes should not be affected.